api/admin/controller.py

```diff
@@ -2451,11 +2451,25 @@ def _get_prior_test_results(self, collection, protocolClass):
         if collection.protocol == OPDSImportMonitor.PROTOCOL:
             protocol = OPDSImportMonitor
 
+        self_test_results = None
         if protocol in provider_apis and issubclass(protocol, HasSelfTests):
             if (collection.protocol == OPDSImportMonitor.PROTOCOL):
-                self_test_results = protocol.prior_test_results(self._db, protocol, self._db, collection, OPDSImporter)
+                extra_args = (OPDSImporter,)
             else:
-                self_test_results = protocol.prior_test_results(self._db, protocol, self._db, collection)
+                extra_args = ()
+            try:
+                self_test_results = protocol.prior_test_results(
+                    self._db, protocol, self._db, collection, *extra_args
+                )
+            except Exception, e:
+                # This is bad, but not so bad that we should short-circuit
+                # this whole process -- that might prevent an admin from
+                # making the configuration changes necessary to fix
+                # this problem.
+                message = _("Exception getting self-test results for collection %s: %s")
+                args = (collection.name, e.message)
+                logging.warn(message, *args, exc_info=e)
+                self_test_results = message % args
```

Comment:
sorry for the retroactive comment but the args should be passed into `_` - I don't think it will work this way.

Reply:
It does seem to work. I get this in the front-end `"Exception getting self-test results for collection manual test: Collection protocol is Manual intervention, but passed into EnkiAPI!"`.

Reply:
Interesting. Maybe the `%` operator is converting it to a string and leaving in the `%s`'s since it has those instead of named args.